### PR TITLE
Add mentoring notes for java/reverse-string

### PR DESCRIPTION
**Pull request by michael-berger-FR**
Already discussed in PR #991

### CODE REVIEW

File: tracks/java/exercises/reverse-string/mentoring.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+### Reasonable solutions
+
+Loop-based solution (see remarks in common suggestions):
+```java
+class ReverseString {
+    String reverse(String inputString) {
+        String output = "";
+        for (int i = inputString.length() - 1; i >= 0; i--) {
+          output += inputString.charAt(i);
```
Comment (sshine) [inline tracks/java/exercises/reverse-string/mentoring.md]:
```suggestion
            output += inputString.charAt(i);
```

Indentation by 4 spaces.

File: tracks/java/exercises/reverse-string/mentoring.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+```
+
+Java8 `IntStream`-based solution:
+```java
+import java.util.stream.IntStream;
+
+class ReverseString {
+    String reverse(String inputString) {
+        final int length = inputString.length();
+        return IntStream.iterate(length-1, i-> i-1).limit(length)
```
Comment (sshine) [inline tracks/java/exercises/reverse-string/mentoring.md]:
```suggestion
        return IntStream.iterate(length-1, i -> i - 1)
                .limit(length)
```

Whitespace in lambda, align all chained methods.

File: tracks/java/exercises/reverse-string/mentoring.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+                .mapToObj(inputString::charAt)
+                .collect(StringBuilder::new, StringBuilder::append, StringBuilder::append).toString();
```
Comment (sshine) [inline tracks/java/exercises/reverse-string/mentoring.md]:
```suggestion
                .collect(StringBuilder::new, StringBuilder::append, StringBuilder::append)
                .toString();
```

Align all chained methods.

**Comment by jmrunkle** [comment]
Not exactly sure how to do this, but I wanted to mention that this PR Fixes #1101

**Comment by sshine** [comment]
@michael-berger-FR: Hello. :-)

It seems that I cannot push to your feature branch.

Would you like to address my suggested changes?

**Comment by sshine** [comment]
@jmrunkle: I think it's okay to not have a corresponding issue for each PR if the discussion is already documented here.

**Comment by sshine** [comment]
I've committed the mentor notes sans whitespace formatting on the premise that they can be fixed later.

**Pull request merged**